astropy/io/misc/qdp.py

```diff
@@ -0,0 +1,451 @@
+
+    Other parameters
+    ----------------
+    input_colnames : List of strings
+        Name of data columns (defaults to ['col1', 'col2', ...]), _not_
+        including error columns.
+
+    Returns
+    -------
+    colnames : List
+        List containing the column names. Error columns will have the name
+        of the main column plus ``_err`` for symmetric errors, and ``_perr``
+        and ``_nerr`` for positive and negative errors respectively
+
+    Examples
+    --------
+    >>> col_in = ['MJD', 'Rate']
+    >>> cols = interpret_err_lines(None, 2, input_colnames=col_in)
+    >>> cols[0]
+    'MJD'
+    >>> err_specs = {'terr': [1], 'serr': [2]}
+    >>> ncols = 5
+    >>> cols = interpret_err_lines(err_specs, ncols, input_colnames=col_in)
+    >>> cols[0]
+    'MJD'
+    >>> cols[2]
+    'MJD_nerr'
+    >>> cols[4]
+    'Rate_err'
+    >>> interpret_err_lines(err_specs, 6, input_colnames=col_in)
+    Traceback (most recent call last):
+        ...
+    ValueError: Inconsistent number of input colnames
+    """
+
+    colnames = ["" for i in range(ncols)]
+    if err_specs is None:
+        serr_cols = terr_cols = []
+
+    else:
+        # I don't want to empty the original one when using `pop` below
+        err_specs = copy.deepcopy(err_specs)
+
+        serr_cols = err_specs.pop("serr", [])
+        terr_cols = err_specs.pop("terr", [])
+
+    if input_colnames is not None:
+        all_error_cols = len(serr_cols) + len(terr_cols) * 2
+        if all_error_cols + len(input_colnames) != ncols:
+            raise ValueError("Inconsistent number of input colnames")
+
+    shift = 0
+    for i in range(ncols):
+        col_num = i + 1 - shift
+        if colnames[i] != "":
+            continue
+
+        colname_root = f"col{col_num}"
+
+        if input_colnames is not None:
+            colname_root = input_colnames[col_num - 1]
+
+        colnames[i] = f"{colname_root}"
+        if col_num in serr_cols:
+            colnames[i + 1] = f"{colname_root}_err"
+            shift += 1
+            continue
+
+        if col_num in terr_cols:
+            colnames[i + 1] = f"{colname_root}_perr"
+            colnames[i + 2] = f"{colname_root}_nerr"
+            shift += 2
+            continue
+
+    assert not np.any([c == "" for c in colnames])
+
+    return colnames
+
+
+def get_tables_from_qdp_file(qdp_file, input_colnames=None):
+    """Get all tables from a QDP file
+
+    Parameters
+    ----------
+    qdp_file : str
+        Input QDP file name
+
+    Other parameters
+    ----------------
+    input_colnames : List of strings
+        Name of data columns (defaults to ['col1', 'col2', ...]), _not_
+        including error columns.
+
+    Returns
+    -------
+    tables : List of `Table` objects
+        List containing all the tables present inside the QDP file
+    """
+
+    contents, ncol = analyze_qdp_file(qdp_file)
+
+    with open(qdp_file) as fobj:
+        lines = list(fobj.readlines())
```

Comment:
`.readlines()` returns a list to begin with on most file-like objects.

Reply:
Oh, I thought it was an iterator. Fixed!